luigi/task.py

```diff
@@ -49,6 +50,30 @@
 _SAME_AS_PYTHON_MODULE = '_same_as_python_module'
 
 
+if sys.version_info < (3, 9):
+    try:
+        from typing_extensions import Annotated, get_args, get_origin, get_type_hints
+    except ImportError:
+        Annotated = get_args = get_origin = get_type_hints = None
+else:
+    from typing import Annotated, get_args, get_origin, get_type_hints
+
+
+def _get_pep539_annotations(tsk_cls):
+    if Annotated is None:
+        return []
+    params = []
+    type_hints = get_type_hints(tsk_cls, include_extras=True)
+    for param_name, annotation in type_hints.items():
+        if get_origin(annotation) is Annotated:
+            args_iter = iter(get_args(annotation))
+            next(args_iter)
+            for metadata in args_iter:
+                if isinstance(metadata, Parameter):
```

Comment:
In case it is not a Parameter, we just ignore the type hint. Shouldn't we instead give feedback to the user at least as a warning telling them that this hint won't be used because it is not valid?